src/profile-logic/graph-color.ts

```diff
@@ -85,6 +86,16 @@ export function getDotColor(color: GraphColor) {
   }
 }
 
+/**
+ * Get the appropriate text color for a graph color background to ensure readability.
+ * Uses mapCategoryColorNameToStyles from colors.ts to get the predefined text colors.
+ */
+export function getTextColor(color: GraphColor): string {
+  // Most GraphColor values exist as keys in STYLE_MAP in colors.ts
```

Comment:
Even though there was a comment I was quite confused when I saw that something of type `GraphColor` was passed to a function with `CategoryColor` in the name. But now I see that mapCategoryColorNameToStyles accepts a generic string and has fallback. Maybe change the comment to the following:

```suggestion
  // There is some overlap between GraphColor and the valid category color
  // values. For other values, mapCategoryColorNameToStyles defaults to gray.
  // This is good enough for now.
```